src/Symfony/Component/HttpClient/ClientState/CurlClientState.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\ClientState;
+
+/**
+ * Internal representation of the cURL client's state.
+ *
+ * @author Alexander M. Turek <me@derrabus.de>
+ *
+ * @internal
+ */
+final class CurlClientState extends AbstractClientState
+{
+    /** @var resource */
+    public $handle;
+    public $pushedResponses = [];
```

Comment:
Can we document the items of that array with `item[]` or better please?

Reply:
Good idea. The documentation would be `array[]`… The `$pushedRespones` structure is in itself a bit complicated. I'll look into this.

Reply:
I've created a class to document the array items.